cluster/clean.sh

```diff
@@ -5,27 +5,33 @@ source ./hack/build/config.sh
+./cluster/kubectl.sh delete ds -l 'operator.cdi.kubevirt.io' -l "cdi.kubevirt.io" -n ${NAMESPACE} --cascade=false --grace-period 0 2>/dev/null || :
 
 # Delete all traces of kubevirt
 namespaces=(default kube-system ${NAMESPACE})
 for i in ${namespaces[@]}; do
-    ./cluster/kubectl.sh -n ${i} delete deployment -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete services -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete validatingwebhookconfiguration -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete secrets -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete configmaps -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete pv -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete pvc -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete ds -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete customresourcedefinitions -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete pods -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete clusterrolebinding -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete rolebinding -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete roles -l 'cdi.kubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete clusterroles -l 'cdikubevirt.io'
-    ./cluster/kubectl.sh -n ${i} delete serviceaccounts -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete deployment -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete services -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete validatingwebhookconfiguration -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete secrets -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete configmaps -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete pv -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete pvc -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete ds -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete customresourcedefinitions -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete pods -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete clusterrolebinding -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete rolebinding -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete roles -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete clusterroles -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
+    ./cluster/kubectl.sh -n ${i} delete serviceaccounts -l 'operator.cdi.kubevirt.io' -l 'cdi.kubevirt.io'
 done
```

Comment:
Removing the CR tells to operator to handle the cleanup.  After it's done, we would go in and cleanup the operator by deleting the operator manifest.

@mhenriks I wonder if there is a way to keep the CR around until the operator give the "all clear" status, when CDI is removed.  Something fancy with finalizers maybe? However, this may be out of scope here and we can work around it in kubevirt-ansible.

Reply:
@aglitke  `make cluster-sync` installs stuff for functional test that the operator does not know about.

Reply:
@rthallisey the operator does exactly that.  The CDI CR will not be deleted until all resources (including stuff created by the cdi controller like clone/upload pods) are deleted